.github/workflows/pull-requests.yaml

```
@@ -0,0 +1,39 @@
+name: Node.js CI 
+
+on:
+  pull_request:
+
+
+jobs:
+  # Needed for grouping check-web3 strategies into one check for mergify
+  build:
+    runs-on: ubuntu-latest
+    needs: build-and-test
+    steps:
+      - run: echo "Done"
+  
+  build-and-test:
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        node:
+          - 'current'
+          - 'lts/*'
+
+
+    steps:
+      - uses: actions/checkout@v4
+      - name: Use Node.js ${{ matrix.node }}
+        uses: actions/setup-node@v4
+        with:
+          node-version: ${{ matrix.node }}
+          cache: 'npm'
+
+      - name: Install dependencies
+        shell: bash
+        run: npm install
+      
+      - name: Linting
+        shell: bash
+        run: npm run format
```

I have no idea if this is going to work, but we can get even simpler:

```suggestion
jobs:
  check-formatting:

    steps:
      - name: Checkout
         uses: actions/checkout@v4

      - name: Install Dependencies
         cache: 'npm'
         run: npm install
         shell: bash

      - name: Test Lint
        run: npm run lint
        shell: bash

      - name: Test Prettier
        run: npm run prettier
        shell: bash
```